internal/agent/agent.go

```diff
@@ -1002,20 +1005,38 @@ func (c *context) ActiveEntitiesChannel() chan string {
 }
 
 func (c *context) SendEvent(event sample.Event, entityKey entity.Key) {
-	if c.eventSender != nil {
-		// limits any string field larger than 4095 chars
-		if c.cfg.TruncTextValues {
-			event = metric.TruncateLength(event, metric.NRDBLimit)
-		}
+	if c.eventSender == nil {
+		aclog.
+			WithField("entity_key", entityKey.String()).
+			Warn("cannot send, event sender not set")
+		return
+	}
 
-		includeSample := c.shouldIncludeEvent(event)
-		if includeSample {
-			if err := c.eventSender.QueueEvent(event, entityKey); err != nil {
-				alog.WithField(
-					"entityKey", entityKey,
-				).WithError(err).Error("could not queue event")
-			}
-		}
+	// truncates string fields larger than 4095 chars
+	if c.cfg.TruncTextValues {
+		orig := event
+		event = metric.TruncateLength(event, metric.NRDBLimit)
+		aclog.
```

Comment:
WDYT if we only log it if `orig != event` ?

Reply:
it will remove noice, sounds good